src/infrastructure/xzdec/BUILD.bazel

```diff
@@ -0,0 +1,33 @@
+load("@aspect_bazel_lib//lib:write_source_files.bzl", "write_source_file")
+load(":wasm.bzl", "wasm_binary")
+
+cc_binary(
+    name = "xzdec",
+    srcs = ["xzdec.c"],
+    linkopts = [
+        "-nostdlib",
+        "-lc",
+        "-Wl,--no-entry",
+    ],
+    deps = ["@xz//:lzma"],
+)
+
+wasm_binary(
+    name = "xzdec_wasm",
+    out = "xzdec.wasm",
+    lib = ":xzdec",
+)
+
+genrule(
+    name = "xzdec_wasm_gz",
+    srcs = [":xzdec_wasm"],
+    outs = ["xzdec.wasm.gz"],
+    cmd = "cat $< | gzip -9 -k -n > $@",
+)
+
+write_source_file(
+    name = "write_xzdec_wasm_gz_to_source_tree",
+    in_file = "xzdec_wasm_gz",
+    out_file = "xzdec.wasm.gz",
+    tags = ["manual"],
```

Comment:
No need for manual here. If the output and the checked in version differ it would be nice if a `test //...` failed.

Reply:
I was confused about this comment because `bazel test //:write_xzdec_wasm_gz_to_source_tree` errors out with `No test targets were found`.

It seems that `write_source_file()` is a macro that generates multiple implicit targets, including a test that uses the `tags=` parameter if set.